BedrockCommand.cpp

```diff
@@ -123,6 +123,14 @@ bool BedrockCommand::areHttpsRequestsComplete() const {
     return true;
 }
 
+map<string,string> BedrockCommand::getCommandLogParams() {
```

Comment:
I don't get why we have this and the whitelist. If we have a global whitelist, what is this for? On the other hand, if we have a per-command function to select params, what's the global whitelist for, shouldn't each command be making that decision?

Reply:
I'm not sure I understand your question. This method adds everything that currently exists in the request into the map. 

The whitelist that exists in `addLogParams` redacts everything that that specific function receives.

For the calls we specifically do for `getCommandLogParams`, we'll pass this array to `addLogParams` so it is correctly redacted, but a bunch of other places will call (indirectly, by calling SALERT/SINFO/etc with a second parameter) `addLogParams` and will need to have other things redacted.

If your point is that we could redact things directly here, I guess we can, but I don't see any gains on doing that.

Reply:
If we are going to use a global whitelist to redact log parameters, I don't see why we need `getCommandLogParams` because the `commandLogParams` are just `command.nameValueMap`. Yes, this adds the command name, but we already have that everywhere we'd want this.

We can just change the two places we call this to do something like:
```
const string logMessage = addLogParams("CRASHING from BedrockServer::sync in command " + command->methodLine, command->nameValueMap);
```

Then we don't need `getCommandLogParams` to make an unnecessary copy of a map we already have.

Reply:
`addLogParams` only receives a map as a second parameter and will only apply the redacting logic if it's part of the map, not if it's part of the text.

The second parameter for `addLogParams` doesn't accept the map we have on `command->request.nameValueMap` since it's a `map<string, SString, STableComp>`

I can either create a new function `addLogParams` that would receive that as a second parameter and iterate through it, or leave the `getCommandLogParams` that is creating a new map.

What do you think its better?

Reply:
the problem with adding a second function with `addLogParams` that can receive an STable is that it will do exactly the same thing we do in `addLogParams`. But maybe that's not a real problem.